internal/meta/base_meta.go

```diff
@@ -555,6 +585,10 @@ func (meta *baseMeta) buildProviderConfig() string {
 	for k, v := range meta.providerConfig {
 		body.SetAttributeValue(k, v)
 	}
+
+	if meta.useAzAPI() {
+		f.Body().AppendNewBlock("provider", []string{"azapi"}).Body()
+	}
```

Comment:
The `azurerm` provider configuration is not needed when using azapi?

Reply:
@wuxu92 The `features` part should also go into the azurerm branch?

Reply:
It appears that AzAPI does not support a `features` block within the `provider` block.

```
│ Error: Unsupported block type
│ 
│   on provider.tf line 2, in provider "azapi":
│    2:   features {}
```